src/lib.rs

```diff
@@ -22,17 +22,47 @@
 
 pub use crate::country::Country;
 pub use crate::error::ValidationError;
-pub use crate::iban::IBan;
 use std::str::FromStr;
 
 pub mod country;
+pub(crate) mod country_specific;
 pub mod error;
-pub mod iban;
 
 #[allow(clippy::all)]
 pub(crate) mod u256 {
     uint::construct_uint! {
-        pub struct U256(4);
+        pub(crate) struct U256(4);
+    }
+}
+
+/// Represents an IBAN and provides helpful methods.
+#[derive(Debug)]
+#[non_exhaustive]
+pub struct IBan {
+    /// The country of this IBAN.
+    pub country: Country,
+    pub(crate) raw: String,
+}
+
+impl IBan {
+    /// Returns the account number of the IBAN.
+    pub fn account_number(&self) -> u128 {
```

Comment:
Why does account_number() returns an u128 and not a String ?
The BBAN are supposed to contain alphanumeric characters, not just numbers